docs/examples/gen_velocities.rst

```diff
@@ -0,0 +1,432 @@
+the system's proper motion) by transforming it to ``lens_frame``.
+
+.. plot::
+    :include-source:
+    :context:
+    
+    v_p_pm = psr_coord.transform_to(lens_frame).velocity.d_z
+
+Effective velocity
+------------------
+
+Combine the velocities of the pulsar, Earth, and the lens into the effective
+velocity
+
+.. math::
+
+    v_\mathrm{eff} = \frac{1}{s} v_\mathrm{ISM}
+                     - \frac{1 - s}{s} ( v_\mathrm{p,orb} + v_\mathrm{p,pm} )
+                     - v_\mathrm{E}
+
+.. plot::
+    :include-source:
+    :context:
+    
+    v_eff = 1. / s * v_ism - ((1 - s) / s) * (v_p_orb + v_p_pm) - v_earth
+
+
+Curvature and scaled effective velocity
+=======================================
+
+The curvature :math:`\eta` can be computed from the effective velocity
+according to
+
+.. math::
+    
+    \eta = \frac{ \lambda^2 d_\mathrm{eff} }{ 2 c v_\mathrm{eff}^2 },
+
+where :math:`\lambda` is the observing wavelength and :math:`c` is the speed of
+light.
+
+.. plot::
+    :include-source:
+    :context:
+
+    lambda_obs = (1400. * u.MHz).to(u.m, equivalencies=u.spectral())
+
+    eta = lambda_obs**2 * d_eff / (2. * const.c * v_eff**2)
+
+Have a look at the curvature at a function of time.
+
+.. plot::
+    :include-source:
+    :context:
+
+    plt.figure(figsize=(12., 6.))
+    
+    quantity_support()
+    time_support(format='iso')
+    
+    plt.plot(t, eta.to(u.s**3))
+
+    plt.ylabel(r'curvature $\eta$ (s$^3$)')
+    
+    plt.show()
+
+Since :math:`v_\mathrm{eff}` can be arbitrarily close to zero (letting
+:math:`\eta` blow up), curvature has a strongly non-uniform prior probability
+distribution (as can be seen from the modulation in amplitude in the figure
+above). For this reason, it is sometimes better to fit for the curvature of the
+secondary spectrum parabola in a space of "scaled effective velocity"
+
+.. math::
+    
+    \frac{ \lambda }{ \sqrt{ 2 \eta c } }
+      = \frac{  \left| v_\mathrm{eff} \right| }{ \sqrt{ d_\mathrm{eff} } }
+
+.. plot::
+    :include-source:
+    :context: close-figs
+    
+    dveff = np.abs(v_eff) / np.sqrt(d_eff)
+    
+Plot this quantity as function of time.
+
+.. plot::
+    :include-source:
+    :context:
+
+    plt.figure(figsize=(12., 6.))
+    
+    quantity_support()
+    time_support(format='iso')
+    
+    plt.plot(t, dveff)
+
+    dveff_lbl = (r'scaled effective velocity '
+                 r'$\frac{ | v_\mathrm{eff} | }{ \sqrt{ d_\mathrm{eff} } }$ '
+                 r'$\left( \frac{\mathrm{km/s}}{\sqrt{\mathrm{pc}}} \right)$')
+    
+    plt.ylabel(dveff_lbl)
+    
+    plt.show()
+    
+To visualize the modulation in scintillation velocity caused by both the
+pulsar's orbital motion and that of the Earth, it can be useful to make a
+2D histogram.
+
+.. plot::
+    :include-source:
+    :context: close-figs
+
+    plt.figure(figsize=(10., 6.))
+
+    plt.hexbin(t.jyear % 1., th_p.value % 1., C=dveff.value,
```

Comment:
Is there a straight-up square 2d histogram option? To me, makes a bit more sense when the two axes are unrelated.

Reply:
I looked into it a few weeks ago and was surprised to find that matplotlib has this `hexbin` function that very quickly makes the 2d histogram we want here, but not a square-pixel equivalent. Maybe I didn't look in the right places.

We could do it manually, but I like that it's in one line of code now, and maybe using hexagonal bins make some sense because we look for structure in this ph_p-ph_e space which also has gradients in diagonal directions.

Reply:
Would it just be `plt.hist2d`?  But don't worry too much, fine to leave as is too.

Reply:
I had another look, and it seems `plt.hist2d` doesn't allow you to pass a "reducer" function that would compute the mean or median of the points that fall within the bin. Here's how I would produce a square-pixel version, giving the mean dveff value for each pixel:
```
x = t.jyear % 1.
y = th_p.value % 1.
nbin = (30, 20)
counts, _, _ = np.histogram2d(x, y, bins=nbin)
sums, _, _ = np.histogram2d(x, y, weights=dveff.value, bins=nbin)
with np.errstate(divide='ignore', invalid='ignore'):
    plt.imshow((sums / counts).T, extent=(0., 1., 0., 1.),
               origin='lower', aspect='auto', interpolation='none')
```
I think I'll leave it as it is.

Reply:
Makes sense!